scripts/area_plot.py

```diff
@@ -0,0 +1,613 @@
+    ----------
+    table : DataFrame
+        A DataFrame with observation on the rows (biom-style table) with 
+        `tax_col` as one of its columns.
+    tax_col : str
+        The column in `table` containig the taxobnomy information
+    tax_delim: str, optional
+        The delimiter between taxonomic groups
+        
+    Returns
+    -------
+    DataFrame
+        The taxonomic strings parsed into n levels
+    """
+    def f_(x):
+        return pd.Series([y.strip() for y in x.split(tax_delim)])
+    
+    return table[tax_col].apply(f_)
+
+
+def level_taxonomy(table, taxa, samples, level, consider_nan=True):
+    """
+    Gets the taxonomy collapsed to the desired level
+    
+    Parameters
+    ----------
+    table : DataFrame
+        A table with observation on the rows (biom-style table) with `samples`
+        in its columns.
+    taxa: DataFrame
+        The taxonomic strings parsed into n levels
+    level: list
+        The level to which the taxonomy should be summarized 
+    samples : list
+        The columns from `table` to be included in the analysis
+    consider_nan: bool, optional
+        Whether the table contains multiple concatenated, in which cases 
+        considering `nan` will filter the samples to retain only the levels 
+        of interest. This is recommended for kracken/bracken tables, but not 
+        applicable for some 16s sequences
+    """
+    level = level.max()
+    print(level)
+    if consider_nan:
+        leveler = (taxa[level].notna() & taxa[(level + 1)].isna())
+    else:
+        leveler = (taxa[level].notna())
+            
+    cols = list(np.arange(level + 1))
+    
+    # Combines the filtered tables
+    level_ = pd.concat(
+        axis=1, 
+        objs=[taxa.loc[leveler, cols],
+              (table.loc[leveler, samples] / table.loc[leveler, samples].sum(axis=0))],
+    #     # sort=False
+    )
+    level_.reset_index()
+    if taxa.loc[leveler, cols].duplicated().any():
+        return level_.groupby(cols).sum()
+    else:
+        return level_.set_index(cols)
+
+
+def profile_one_level(collapsed, level, threshhold=0.01, count=8):
+    """
+    Gets upper and lower tables for a single taxonmic level
+    
+    Parameters
+    ----------
+    Collapsed: DataFrame
+        The counts data with the index as a multi-level index of 
+        levels of interest and the columns as samples
+    threshhold: float, optional
+        The minimum relative abundance for an organism to be shown
+    count : int, optional
+        The maximum number of levels to show for a single group
+        
+    Returns
+    -------
+    DataFame
+        A table of the top taxa for the data of interest
+    """
+    collapsed['mean'] = collapsed.mean(axis=1)
+    collapsed.sort_values(['mean'], ascending=False, inplace=True)
+    collapsed['count'] = 1
+    collapsed['count'] = collapsed['count'].cumsum()
+
+    thresh_ = (collapsed['mean'] > threshhold) & (collapsed['count'] <= count)
```

Comment:
```suggestion
    thresh_ = (collapsed['mean'] > threshold) & (collapsed['count'] <= count)
```

Reply:
Lots of unnecessary trailing underscores. Please search the file for `_ ` and reconsider them.

Reply:
I like trailing underscores in my variable names.